### ISSUE TITLE
Contribute to tflearn

### ISSUE THREAD

**Issue by FreakTheMighty** [MarvinTeichmann/tensorflow-fcn#1]
@MarvinTeichmann thank you for this implementation of FCN in tensorflow. This is exactly what I was looking for. I'd like to humbly suggest you port your work into [tflearn](https://github.com/tflearn/tflearn). I think that your implementation of `_upscore_layer` as well as the loss function you've defined would be very useful in tflearn.

**Comment by MarvinTeichmann** [comment]
Thanks for the hint and good idea. I have implemented the [upscore_layer](https://github.com/MarvinTeichmann/tflearn/tree/FCN_upscore)  and the [semantic segmentation loss](https://github.com/MarvinTeichmann/tflearn/tree/semantic_segmentation_loss) to tflearn. Can you have a short look at it? I have no experience with tflearn. Needed to do some changed to fit in to the framework but have cannot test whether it works now. 

Thanks in advance! If you approve my code, I will open a pull request.

**Comment by FreakTheMighty** [comment]
@MarvinTeichmann wow, that was fast :D. I can take a closer look tonight, but I think it makes sense to open a pull request sooner rather than later. The maintainer, @aymericdamien, of tflearn will be able to give the most substantive feedback.

I only get a few hours a day to work on these things, but I'd like to contribute an example that recreates the FCN using tflearn and your new layers and loss.

**Comment by aymericdamien** [comment]
@MarvinTeichmann that indeed looks really great! It would definitely be useful for our users. Do not hesitate to send us a pull request :) Just my suggestion is to move the 'semantic segmentation loss' to 'objectives.py' file instead of 'losses.py' file, and change 'logits'/'labels' arguments name to 'y_pred'/'y_true', for consistency with other objective functions.

@FreakTheMighty You can go ahead, thanks!

**Comment by FreakTheMighty** [comment]
@aymericdamien I've started working on porting @MarvinTeichmann networks here. https://github.com/MarvinTeichmann/tflearn/pull/1 I'm  sure I will get stuck along the way, but I think it will be good experience.

One thing I noticed about the `weak_cross_entropy_2d_loss` objective is that it takes  more than two arguments where tflearn objectives are passed as strings into the solver. How would that interface work in tflearn?

**Comment by aymericdamien** [comment]
Yes, because of the extra argument, it cannot directly be called as a string argument in `regression` layer, however, loss argument can be a function, so it could be something like:

```
def my_loss(y_pred, y_true):
  return tflearn.weak_cross_entropy_2d_loss(y_pred, y_true, num_classes=10)
net = ...
net = tflearn.regression(net, loss=my_loss)
```

Another approach would be to remove num_classes argument and calculate it automatically by checking dimension of y_true. So passing just 

```
net = ...
net = tflearn.regression(net, loss='weak_cross_entropy_2d_loss')
```

will work

**Comment by MarvinTeichmann** [comment]
Merged in tflearn/tflearn#137.
